src/server.js

```diff
@@ -233,6 +231,14 @@ module.exports = function (options = {}) {
 
 		process.on("SIGINT", exitGracefully);
 		process.on("SIGTERM", exitGracefully);
+		process.on("SIGHUP", function () {
+			const hasChanges = Helper.reloadConfig();
```

Comment:
My above comments would transform this to `const hasChanges = Helper.doesConfigHaveChangs()` which would call `reloadConfig()`